<?xml version="1.0"?>
<launch>

  <!-- manual controller for Jackal -->
<!--  <node name="joy_ros" pkg="joy" type="joy_node">-->
<!--      <param name="/dev" value="/dev/input/js0"/>-->
<!--  </node>-->
<!--  <node name="ctrl_joy" pkg="ieee_uav" type="mavros_joy_controller.py"/>-->

  <param name="/altitude_fixed" value="0.8"/>
  <param name="/c_x" value="320.5"/>
  <param name="/c_y" value="240.5"/>
  <param name="/f_x" value="319.99"/>
  <param name="/f_y" value="319.99"/>

  <param name="/pcl_max_range" value="20.0"/>
  <param name="/depth_topic" value="/d435i/depth/image_raw"/>
  
  <param name="/depth_base" value="d435i/depth_camera_link"/>
  <param name="/body_base" value="base_link"/>
  <param name="/fixed_frame" value="map"/>

  <!-- For target prediction -->
  <param name="/target_traj_hz" value="30.0"/>
  <param name="/traj_leng" value="30"/>
  <param name="/traj_leng_past" value="300"/>
  <!-- <param name="/target_predict_seg" value="1"/> -->

  <!-- Nodes  -->

  <!-- main code: traversability check, obstacle check, trajectory generate, select  -->
  <node name="ieee_uav" pkg="ieee_uav" type="ieee_uav_node" output="screen"/>
                      <!-- launch-prefix="gdb -ex run --> <!--args" -->

  <!-- High-level control code (MPC) -->
   <node name="mpc_control" pkg="ieee_uav" type="mpc-slsqp.py"/> 

  <!-- YOLO -->
  <param name="/bbox_out_topic" value="/bboxes"/>
  <node name="yolo_ros_simple_node" pkg="yolo_ros_simple" type="ros_opencv_dnn.py" output="screen">
    <param name="inference_rate" value="15"/>
    <param name="img_in_topic" value="/d435i/depth/rgb_image_raw"/>
    <param name="img_out_topic" value="/detected"/>
    <param name="confidence_threshold" value="0.8"/>
    <param name="nms_threshold" value="0.4"/>
    <param name="class_file" value="$(find yolo_ros_simple)/scripts/obj.names"/>
    <param name="weight_file" value="$(find yolo_ros_simple)/scripts/yolov4-tiny-3l-uav_final.weights"/>
    <param name="cfg_file" value="$(find yolo_ros_simple)/scripts/yolov4-tiny-3l-uav.cfg"/>
  </node>

  <!-- tf publisher  -->
  <node name="tf_broadcaster" pkg="ieee_uav" type="tf_broadcaster.py"/>

  <!-- GT to path: for visualization  -->
  <node name="gt_visualizer" pkg="ieee_uav" type="visualizer.py"/>


</launch>
